linode/manager
# upcoming: [DI-19819] - Added a filter builder component for ACLP that builds the filters per service type

## Description 📝
The CloudPulseDashboardFilterBuilder component builds the required mandatory global filters from a JSON config based on service types like linode, dbass etc and the CloudPulseDashboard Landing component pushes the selected filters to the dashboard component

## Changes  🔄
List any change relevant to the reviewer.
- Added a new CloudPulseDashboardFilterBuilder that will build the global filters based on FilterConfig.ts per service type
- Added a new CloudPulseComponentRenderer component that builds the react element based on filterKey
- Added UT's for both of the component
- Complete integration of the CloudPulseDashboardFilterBuilder in the GlobalFilters.tsx
- Added logic in the CloudPulseDashboardLanding component to pass the filters selected to the dashboard component
- Changes for moving CloudPulseTimeRangeSelect from **select to Autocomplete**
- CSS changes making the filters compatible with different screens like desktop, tablet and mobile.
- The filter builder component, shows a scroll bar after two rows
- Added a new monitor icon svg as per UX mockups

## Target release date 🗓️
13th August 2024

## Preview 📷
**Include a screenshot or screen recording of the change**

:bulb: Use `<video src="" />` tag when including recordings in table.

| Before  | After   |
| ------- | ------- |
|![Screenshot 2024-07-26 at 3 22 24 PM](https://github.com/user-attachments/assets/38259036-72d7-4296-8ab1-3f15fe28090e)|![Screenshot 2024-08-07 at 6 18 53 PM](https://github.com/user-attachments/assets/bd3f4495-fbe5-4a2c-85a3-32b947fdd21a)|
|![Screenshot 2024-08-07 at 6 20 40 PM](https://github.com/user-attachments/assets/c914a79f-fa6e-49fa-a71a-189b604520dc)|![Screenshot 2024-08-07 at 6 20 52 PM](https://github.com/user-attachments/assets/8a26a55e-0972-4bcc-a9f0-d744326c2dfc)|
|![Screenshot 2024-08-07 at 6 21 29 PM](https://github.com/user-attachments/assets/f666587b-fa2e-456e-b91a-d57683507cfd)|![Screenshot 2024-08-07 at 6 21 39 PM](https://github.com/user-attachments/assets/8375b846-ecfa-44a4-8ce6-0b8a8827f87d)|
|![Screenshot 2024-08-07 at 6 22 11 PM](https://github.com/user-attachments/assets/181c1bb2-35a4-4a6e-b819-f626b8491863)|![Screenshot 2024-08-07 at 6 22 33 PM](https://github.com/user-attachments/assets/519a1001-3b5a-47ab-b195-3bddf1cf3493)|
|![Screenshot 2024-08-07 at 6 40 47 PM](https://github.com/user-attachments/assets/7d32233a-e885-4d32-8f2d-a7da3117653a)|![Screenshot 2024-08-07 at 6 42 59 PM](https://github.com/user-attachments/assets/cf15804c-e24e-4586-a5c7-cf2ea40e70a1)|


## How to test 🧪

1. Enable mock service as some endpoints are not ready to use
2. Go the Monitor Landing Page
4. Select a Dashboard from dashboard selection dropdown
5. You will be able to see the filters build below the existing global filters
6. After selecting all the mandatory filters, you will able to see the widgets being rendered

## As an Author I have considered 🤔

*Check all that apply*
